sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -351,4 +352,28 @@ public static Duration getDefaultTimeoutFromEnvironment(Configuration configurat
             return defaultTimeout;
         }
     }
+
+    /**
+     * Merges two {@link Context Contexts} into a new {@link Context}.
+     *
+     * @param into Context being merged into.
+     * @param from Context being merged.
+     * @return A new Context that is the merged Contexts.
+     * @throws NullPointerException If either {@code into} or {@code from} is null.
+     */
+    public static Context mergeContexts(Context into, Context from) {
+        Objects.requireNonNull(into, "'into' cannot be null.");
+        Objects.requireNonNull(from, "'from' cannot be null.");
+
+        Context[] contextChain = from.getContextChain();
+
+        Context returnContext = into;
+        for (Context toAdd : contextChain) {
+            if (toAdd != null) {
+                returnContext = returnContext.addData(toAdd.getKey(), toAdd.getValue());
```

Comment:
Any reason why we aren't using the `into` here directly?

Reply:
Nope, just added the additional variable to make it clearer